lib/jsdom/living/geometry/DOMMatrixReadOnly-impl.js

```diff
@@ -0,0 +1,519 @@
+/* eslint-disable no-use-before-define */
+"use strict";
+const DOMException = require("domexception");
+const {
+  matrix3dArrayProperties,
+  matrix2dArrayProperties
+} = require("../helpers/geometry/matrix");
+const {
+  validateDOMMatrixInit,
+  arrayFromDOMMatrixInit
+} = require("../helpers/geometry/dommatrix-init");
+const { parseDOMStringToMatrixFunctions } = require("../helpers/geometry/matrixDomStringParser");
+
+class DOMMatrixReadOnlyImpl {
+  constructor(constructorArgs = []) {
+    let [matrixInit] = constructorArgs;
+
+    if (
+      typeof matrixInit === "undefined" ||
+      matrixInit === "" ||
+      matrixInit === "none"
+    ) {
+      matrixInit = [1, 0, 0, 1, 0, 0];
+    } else if (typeof matrixInit === "string") {
+      try {
+        matrixInit = this.parseDomString(matrixInit);
+      } catch (e) {
+        throw new
+        DOMException(`Failed to construct ${this.constructor.name}: failed to parse ${matrixInit}`, "SyntaxError");
+      }
+    }
+
+    /**
+     * Note: An underscore indicates that the property is defined by a property accessor.
+     * These properties produce side effects upon getting or setting.
+     * To prevent a ton of property accessors, only those that produce side effects are defined as property accessors.
+     */
+    this.m11 = 0;
+    this.m12 = 0;
+    this._m13 = 0;
+    this._m14 = 0;
+    this.m21 = 0;
+    this.m22 = 0;
+    this._m23 = 0;
+    this._m24 = 0;
+    this._m31 = 0;
+    this._m32 = 0;
+    this._m33 = 0;
+    this._m34 = 0;
+    this.m41 = 0;
+    this.m42 = 0;
+    this._m43 = 0;
+    this._m44 = 0;
+    this._is2D = false;
+
+
+    if (matrixInit.length === 6) {
+      this._initialize2dMatrix(matrixInit);
+    } else if (matrixInit.length === 16) {
+      this._initialize3dMatrix(matrixInit);
+    } else {
+      throw new TypeError(`Invalid length for constructor init, expected 6 or 16, but received ${matrixInit.length}`);
+    }
+  }
+
+  parseDomString(domstring) {
```

Comment:
This looks to be private, i.e., not exposed to the web. Let's prefix the name with an underscore.

Additionally, this seems to implement https://drafts.fxtf.org/geometry/#parse-a-string-into-an-abstract-matrix. Link it, and rename the function to make it clear that is the case.